src/database.ts

```diff
@@ -1,15 +1,20 @@
 import { PrismaClient } from '@prisma/client';
+import { getModuleBinding, getChildLogger } from './logger';
+import { dbLogLevels } from './env';
 
+const dbLogger = getChildLogger({ msgPrefix: 'DATABASE' }, { module: getModuleBinding(__filename) });
 let prisma: PrismaClient;
 
 export const getPrisma = () => {
+  dbLogger.debug('Getting prisma client');
   if (!prisma) {
     prisma = new PrismaClient({
       datasources: {
         db: {
           url: 'file:./db.sqlite',
         },
       },
+      log: dbLogLevels,
```

Comment:
i dont get this q.

Are you asking if pino prints it? or something else?

Reply:
The log of the Prisma is not handled by pino, but it's rather just printed into the console. So if we to add a `transport` to pino, prisma logs will not end up there, right?

Reply:
as per https://github.com/prisma/prisma/issues/10475 the logging with custom client is not yet implemented for prisma. Would you like me to extend the prisma client with custom logger as per the example in the same linked thread?

Reply:
I've had to rewrite a bunch of `database.ts` lines:

  - i dont see the point in the getter: we can import prisma directly and there will be just a single instance of it anyways. I don't see the reason to import from `database` unless you want to access prisma, so building it first time in-place when it actually is required does not seem like a big win to me.
  - if we declare the prisma without instantiating it immediately then it derives and keeps its generics. Hence, all the event generics are already defined when we would do `prisma.$on`. With derived generics it forbids using `$on('query')` because type system thinks that we did not define logging event `'query'`. I.e. factual instance of prisma is more specific than the one we declare which blocks us from event handling.

Reply:
> we can import prisma directly and there will be just a single instance of it anyways

Can you please link a proof that no extra connection is opened when you import it every time. I think it's extra critical for the sqlite since it just doesn't work with multiply connections

Reply:
https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#prismaclient-in-long-running-applications

the official recommendation

https://stackoverflow.com/questions/46937494/es6-variable-import-by-reference-or-copy

stackoverflow answer with another reflink in it